### DISCUSSION TITLE
Persistent BT Pairing Requests

### DISCUSSION THREAD

**Discussion by bsw-del** [Navid200/xDrip#82] [Dexcom]
Hey there,
Just switched phones and am now experiencing frequent bluetooth pairing requests. Happens ~ every 30 mins - 60 mins. Seems to be correlated to the phone not being in range or being challenged to make a connect to the transmitter, this was never an issue on my previous phone. Logs in the screenshot. 
Followed guide to get settings correct. Am running latest nightly version. Device is Pixel 6 - Android 12

![Screenshot_20211027-095333](https://user-images.githubusercontent.com/57199290/139084566-f35d1f5f-3f5c-4ff7-88b9-f911df635b62.png)

Any help would be greatly appreciated, I've searched extensively and haven't yet found a solution as the new version of xdrip+ nicely dealt with this previous problem (went through that with you all last year. :) )

**Comment by Navid200** [comment]
Would you please post a screenshot of the two system status pages (classic and G5/G6)?
Have you disabled battery optimization of xDrip on the new phone?

**Reply by bsw-del** [reply]
Thanks @Navid200 for the rapid response. 
Yep battery optimization is disabled. I've noticed that on G5/G6 page, the Bluetooth Link will change to "Unpaired" in the few minutes before it requests it's next connection, and that is when it will ask to pair again.

![Screenshot_20211027-103110](https://user-images.githubusercontent.com/57199290/139087304-44979889-3355-4730-9545-efb0c86c9c99.png)
![Screenshot_20211027-103113](https://user-images.githubusercontent.com/57199290/139087311-4132e067-a98f-44e4-a500-ec78ed534cdd.png)
![Screenshot_20211027-103259](https://user-images.githubusercontent.com/57199290/139087314-4c49085a-7f83-40f9-b65f-0c54dfe475bc.png)
.

**Comment by Navid200** [comment]
Would you please show the following page?
Settings -> G5/G6 Debug Settings

**Reply by bsw-del** [reply]
Please see below
![Screenshot_20211027-104116](https://user-images.githubusercontent.com/57199290/139088578-719b5fc1-9b73-45bf-8638-db803cc624f9.png)

**Comment by Navid200** [comment]
Would you please disable "Avoid Scanning" only?
This is just an experiment.

Would you please tell me if that makes the disconnects worse or if it reduces the frequency of disconnects?

**Reply by bsw-del** [reply]
Will do. Will report back in a bit once I get a sense for what it does.

**Comment by Navid200** [comment]
Would you please show?

Android Settings -> Battery

**Comment by Navid200** [comment]
This is a brand new transmitter.  Is it a coincidence that you have a brand new transmitter and a brand new phone exactly at the same time?
Or, did you use this same phone with the previous transmitter yesterday?

**Reply by bsw-del** [reply]
Not a coincidence. Tried this phone with a previous transmitter yesterday. Couldn't get it working (same issue) and in troubleshooting switched to a new one to see if that would resolve, thought maybe it was trying to pair with the old phone. 
![Screenshot_20211027-105318](https://user-images.githubusercontent.com/57199290/139091771-30f45596-ff60-4e0e-940d-59ead60af349.png)

**Comment by bsw-del** [comment]
Since avoid scanning and reboot.. no pairing request yet.

**Comment by Navid200** [comment]
Under battery, what's under Adaptive Preferences?

**Comment by bsw-del** [comment]
![Screenshot_20211027-110208](https://user-images.githubusercontent.com/57199290/139092638-3444343c-489b-46a5-8b55-d41acd365a23.png)

Same settings as on my previous phone. Which was fine..

**Comment by bsw-del** [comment]
Take it back, I just got a pairing request again.
![Uploading Screenshot_20211027-110421.png…]()

**Comment by Navid200** [comment]
Please disable Adaptive Battery.

**Reply by bsw-del** [reply]
Ok will do. 
I did just re-check, my previous device was on same OS (Android 12) and adaptive battery was on, and was not causing any issue. But happy to try it out!

**Comment by bsw-del** [comment]
![Screenshot_20211027-110421](https://user-images.githubusercontent.com/57199290/139093471-5401de76-3ecf-4ab2-a49d-56ff48414765.png)

**Comment by bsw-del** [comment]
Adaptive battery off. Just had a pairing request. 
Log below
![Screenshot_20211027-112255](https://user-images.githubusercontent.com/57199290/139096565-48dd2eb7-bc51-4151-a801-778271d02ccf.png)

**Comment by Navid200** [comment]
What happens if you also disable "Minimize Scanning"?

**Reply by bsw-del** [reply]
Done and will monitor. 
Thanks again for this, super helpful. Any hunches at this point?
Logs suggest anything, or too generic?

**Reply by Navid200** [reply]
There is an existing issue:
https://github.com/NightscoutFoundation/xDrip/issues/1664

But, I prefer if this is resolved without using that workaround.
That's why I work with everyone experiencing this before asking them to try that workaround.

**Reply by bsw-del** [reply]
Awesome. Thanks. 
So far it hasn't asked to pair in the past 40 minutes.

**Reply by bsw-del** [reply]
Thanks @Navid200 for the link, i changed settings as per Jamorham and it is working ok now. I'm going to leave it like this for a bit because I prefer to have my battery optimization on. 
Thanks a mill for the support today!

**Comment by starbuck93** [comment]
Pixel 6 Pro here on Android 12, same issue. I switched phones in the middle of this transmitter from my, previously bonded, Pixel 3XL on Android 12. I assumed, before reading about it, that the transmitter just expects to see the P3XL instead of the P6Pro and complains about it. I did not try to disable collection inside xDrip in the P3XL first before pairing with the P6Pro. At this time, I'm getting a bunch of "Could not reply to auth challenge" messages but data is coming in just fine with "Allow OB1 Unbonding" unchecked. I also assume that when it's time to pair a new transmitter in about 70 days that I'll be able to check that box again and it'll be fine.

**Reply by Navid200** [reply]
You can ignore those logs as long as you are getting consistent readings.

**Reply by realmajortom** [reply]
Chiming in to add that I experienced this same issue after upgrading from Pixel 5 to Pixel 6 Pro. Also solved it the same way (disallowing OB1 unbonding).

@Navid200 please let me know if you'd like me to help investigate or test further.

**Comment by loop2016** [comment]
Pixel 6 here on Android 12. I also had the pairing request issue and also frequent disconnects with the recommended settings, but after disabling the "adaptive battery" and disallowing OB1 unbonding" everything seems to work better than ever. Thank you so much for your efforts!
